cmake/extensions.cmake

```diff
@@ -20,17 +20,29 @@ function(get_board_without_ns_suffix board_in board_out)
   endif()
 endfunction()
 
-# Add a kconfig overlay file to a child image.
+# Append a value to a variable prefixed with a child image name.
+# This can be used by a parent image to modify the variables passed to its
+# child images. This must be invoked before 'add_child_image(image)'
+function(child_image_append_conf image val type)
```

Comment:
I'm not sure we should document this function.

I don't like such generic function, as people might start to use it for unintended purposes, and thus makes it even harder for our users to understand when / why / where settings are being defined.

We should use the common `boards/` and `child_image` config options as much as possible, and extend those if need be.

I know there is a need for the current `add_overlay_config` function and maybe even a `devicetree` overlay also if there really is a need there. (preference would still be a extension of: http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_multi_image.html#image-specific-variables to support devicetree overlay if we have a need)

If keeping this helper function, then clearly mark it `Internal use only`.